java/ql/src/semmle/code/java/frameworks/ApacheHttp.qll

```diff
@@ -13,3 +13,23 @@ class ApacheHttpEntityGetContent extends Method {
     this.getName() = "getContent"
   }
 }
+
+/**
+ *  A class derived from the `HttpRequestBase` or the `BasicHttpRequest`
+ *  class of the Apache Http Client `org.apache.http` library
+ */
+class TypeApacheHttpRequestBase extends RefType {
+  TypeApacheHttpRequestBase() {
+    this
+        .getASourceSupertype*()
+        .hasQualifiedName("org.apache.http.client.methods", "HttpRequestBase") or
+    this.getASourceSupertype*().hasQualifiedName("org.apache.http.message", "BasicHttpRequest")
+  }
+}
+
+/* A class representing the `RequestBuilder` class of the Apache Http Client library */
```

Comment:
Qldoc is missing an additional `*` to make it proper qldoc rather than a simple comment. Also, drop the "A class representing .." style of qldoc - all ql classes represent something, so we might as well state directly what that something is.